nightfall-client/src/services/commitment-sync.mjs

```diff
@@ -30,11 +30,23 @@ export async function decryptCommitment(transaction, zkpPrivateKey, nullifierKey
         transaction.recipientAddress, // It contains the tokenID encrypted (which is a field)
         ...transaction.compressedSecrets,
       ];
-      const [packedErc, unpackedTokenID, ...rest] = decrypt(
-        generalise(key),
-        generalise(edwardsDecompress(transaction.tokenId)), // Compressed public key is stored in token ID
-        generalise(cipherTexts),
-      );
+      let packedErc;
+      let unpackedTokenID;
+      let rest;
+      if (!REGULATOR_URL) {
+        [packedErc, unpackedTokenID, ...rest] = decrypt(
+          generalise(key),
+          generalise(edwardsDecompress(transaction.tokenId)), // Compressed public key is stored in token ID
+          generalise(cipherTexts),
+        );
+      } else {
+        [packedErc, unpackedTokenID, ...rest] = decrypt(
+          generalise(key),
+          generalise(edwardsDecompress(transaction.tokenId)), // Compressed public key is stored in token ID
+          generalise(cipherTexts),
+          BigInt(transaction.value),
```

Comment:
I'm not sure if the transaction value is a good candidate for a being a salt/nonce. We'd need something more random. The transaction values in Nightfall are all integers, right? IMHO, they don't provide a good source of randomness.

Reply:
We store a random value in `transaction.value`. It's not the value itself of the transaction. In transfers this value is always 0 now. That's why we have taken this variable to store the nonce. Like in other parts to optimize the size of the information stored in the transaction and don't create extra fields which will increment the gas per transaction in calldata.